From 2618ab101f5faaa887c9feeeee84a7d8bc97cbda Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Wed, 16 Nov 2022 11:27:24 +0100
Subject: [PATCH] meson.build: correctly handle disabled ssh_agent option

Existing code produces these errors:
| gcr/meson.build:61:0: ERROR: Unable to get the path of a not-found external program
| gcr/meson.build:101:5: ERROR: Unknown variable "ssh_add_path".

Signed-off-by: Alexander Kanavin <alex@linutronix.de>

Upstream-Status: Backport [ https://gitlab.gnome.org/GNOME/gcr/-/commit/9b67bb18d8409e0e693cc6000507acbd73a30eab ]

---
 meson.build | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index dd890a9..1b17b18 100644
--- a/meson.build
+++ b/meson.build
@@ -82,8 +82,10 @@ conf.set('HAVE_TIMEGM', cc.has_function('timegm'))
 conf.set('HAVE_MLOCK', cc.has_function('mlock'))
 conf.set_quoted('GPG_EXECUTABLE', gpg_path)
 conf.set_quoted('LIBGCRYPT_VERSION', libgcrypt_dep.version())
-conf.set_quoted('SSH_ADD_EXECUTABLE', ssh_add_path)
-conf.set_quoted('SSH_AGENT_EXECUTABLE', ssh_agent_path)
+if get_option('ssh_agent')
+  conf.set_quoted('SSH_ADD_EXECUTABLE', ssh_add_path)
+  conf.set_quoted('SSH_AGENT_EXECUTABLE', ssh_agent_path)
+endif
 conf.set10('WITH_SYSTEMD', with_systemd)
 config_file = configure_file(
   output: 'config.h',
